### PR TITLE
Feature/support esp32h2

### PR DESCRIPTION
**Pull request by adwait-esp**
- Ported the changes from esptools.py for esp32h2 chipset.
- Tested with H2 chipset and able to flash firmware image on h2 using the tool.

### CODE REVIEW

**Review by igrr — APPROVED** [review]
LGTM! Just one lint issue to fix so that pipeline passes.

File: src/esploader.ts [hunk 1]
```diff
@@ -18,6 +18,10 @@ async function magic2Chip(magic: number): Promise<ROM | null> {
       const { ESP32C6ROM } = await import("./targets/esp32c6");
       return new ESP32C6ROM();
     }
+    case 0xD7B73E80:{
```
Comment (brianignacio5) [inline src/esploader.ts:21]: `case 0xd7b73e80`

**Review by brianignacio5 — CHANGES_REQUESTED** [review]
LGTM with small change

File: src/esploader.ts (hunk 1 above)
Comment (adwait-esp) [reply src/esploader.ts]: Fixed

**Comment by adwait-esp** [comment]
Merging since the minor review comments are addressed and CI pipeline job has succeeded as well

**Pull request merged**